src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -10477,5 +10470,3000 @@ public class Class
+                    public void M(ImmutableArray<int arr);
+                    """;
+
+                UsingDeclaration(source, options,
+                    // (1,34): error CS1003: Syntax error, '>' expected
+                    // public void M(ImmutableArray<int arr);
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "arr").WithArguments(">").WithLocation(1, 34)
+                    );
+
+                N(SyntaxKind.MethodDeclaration);
+                {
+                    N(SyntaxKind.PublicKeyword);
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.VoidKeyword);
+                    }
+                    N(SyntaxKind.IdentifierToken, "M");
+                    N(SyntaxKind.ParameterList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.Parameter);
+                        {
+                            N(SyntaxKind.GenericName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "ImmutableArray");
+                                N(SyntaxKind.TypeArgumentList);
+                                {
+                                    N(SyntaxKind.LessThanToken);
+                                    N(SyntaxKind.PredefinedType);
+                                    {
+                                        N(SyntaxKind.IntKeyword);
+                                    }
+                                    M(SyntaxKind.GreaterThanToken);
+                                }
+                            }
+                            N(SyntaxKind.IdentifierToken, "arr");
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        [WorkItem(48566, "https://github.com/dotnet/roslyn/issues/48566")]
+        public void MissingClosingAngleBracket_MethodArgumentList02()
+        {
+            foreach (var options in new[] { TestOptions.Script, TestOptions.Regular })
+            {
+                const string source =
+                    """
+                    public void M(ImmutableArray<int arr, ImmutableArray<int> another);
+                    """;
+
+                UsingDeclaration(source, options,
+                    // (1,34): error CS1003: Syntax error, ',' expected
+                    // public void M(ImmutableArray<int arr, ImmutableArray<int> another);
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "arr").WithArguments(",").WithLocation(1, 34),
+                    // (1,59): error CS1003: Syntax error, '>' expected
+                    // public void M(ImmutableArray<int arr, ImmutableArray<int> another);
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "another").WithArguments(">").WithLocation(1, 59)
+                    );
```

Comment:
hrmm... what happened here?  seems like not a great recovery.

Reply:
We are expecting
```csharp
public void M(ImmutableArray<int, arr, ImmutableArray<int>> another);
```
in this scenario. This is indeed somewhat bad, but we are not scanning the entire parameter list to figure out the best error recovery. It only looks ahead by one token to see if it is a closing kind. I don't know if the current description of the PR covers this explanation, will pass through it.